### PR TITLE
Create operational resources

### PR DESCRIPTION
**Pull request by LukeWood**
This commit begins implementing the operational pieces
of the keras-cv repo.  This includes a setup.py package,
a github action to lint, and scripts to complete linting.

### CODE REVIEW

**Comment by LukeWood** [comment]
@fchollet @qlzh727 let me know what you think!

File: shell/lint.sh [hunk 1]
```diff
@@ -0,0 +1,27 @@
+#!/bin/bash
```
Comment (bhack) [inline shell/lint.sh:1]: Are you interested to integrate pre-commit hooks directly as I've proposed in https://github.com/tensorflow/tensorflow/pull/48371

Comment (LukeWood) [reply shell/lint.sh:1]: yeah, do I need to remove the shebang?  I noticed it was absent in the other repos, I assume this is for good reason?

Comment (LukeWood) [reply shell/lint.sh:1]: Possibly once we hit a certain level of complexity.   We're hoping to standardize across keras-cv, keras-nlp, and keras-tuner. 

@fchollet @qlzh727 thoughts?

Comment (bhack) [reply shell/lint.sh:1]: If you check my PR relies on https://github.com/pre-commit/pre-commit that It could help a bit also to maintain repos in sync on the same linting and also with the local developer setup.

Comment (bhack) [reply shell/lint.sh:1]: P.s. generally this could reduce triagers  roundtrip request for linting on CI fail.

Comment (LukeWood) [reply shell/lint.sh:1]: Let's open a Github issue for this!  I think this is a good idea but at the moment I lack the cycles to do a full proper CI implementation.  Let's use the minimal version for now and revisit in the near future when we have some real code to test on etc

Comment (bhack) [reply shell/lint.sh:1]: I can have some cycles.. but please open a new ticket on your side with a contribution welcome label so I know that you have evaluated the approach  before I prepare a PR

Thanks

Comment (fchollet) [reply shell/lint.sh:1]: If you're able to open a PR, please do -- that sounds super helpful 👍

File: .gitignore [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+*.swp
+.idea
+*.pyc
+.pytest_cache
+*.egg-info
+
+#VS Code files and container
+.vscode/
+.devcontainer/
+
```
Comment (fchollet) [inline .gitignore:11]: Basically everything below this line can be deleted

File: setup.py [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+# limitations under the License.
+
+"""Setup script."""
+
+from setuptools import find_packages
+from setuptools import setup
+
+setup(
+    name="keras-cv",
+    description="Computer Vision extensions for Keras.",
```
Comment (fchollet) [inline setup.py:22]: "Industry-strength computer vision"

File: shell/lint.sh (hunk 1 above)
Comment (bhack) [reply shell/lint.sh:1]: Ok when this is in master I will try to prepare a PR.

File: setup.py [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+    url="https://github.com/keras-team/keras-cv",
+    author="The KerasCV authors",
+    author_email="keras-cv@google.com",
+    license="Apache License 2.0",
+    # tensorflow isn't a dependency because it would force the
+    # download of the gpu version or the cpu version.
+    # users should install it manually.
```
Comment (fchollet) [inline setup.py:29]: I think this comment is no longer applicable, we should be able to list TF as a dependency

File: setup.py [hunk 5]
```diff
@@ -0,0 +1,61 @@
...
+    install_requires=[
+        "packaging",
+        "numpy",
+        "requests",
+        "scipy",
+        "tensorboard",
+        "ipython",
+        "kt-legacy",
```
Comment (fchollet) [inline setup.py:37]: We should just list the stuff we actually use, as we go. This list should start with just `["tensorflow"]`. Numpy/Keras are automatically included when we depend onTF.

File: setup.py [hunk 6]
```diff
@@ -0,0 +1,61 @@
...
+    ],
+    extras_require={
+        "tests": [
+            "pytest",
+            "flake8",
+            "isort",
+            "black",
```
Comment (fchollet) [inline setup.py:44]: This setup seems to only use pytest, flake8, and black, no?

File: setup.py [hunk 7]
```diff
@@ -0,0 +1,61 @@
...
+"""Setup script."""
+
+from setuptools import find_packages
+from setuptools import setup
+
+setup(
+    name="keras-cv",
+    description="Computer Vision extensions for Keras.",
+    url="https://github.com/keras-team/keras-cv",
+    author="The KerasCV authors",
```
Comment (fchollet) [inline setup.py:24]: "Keras team"

Comment (LukeWood) [reply setup.py:24]: done

File: setup.py (hunk 6 above)
Comment (LukeWood) [reply setup.py:44]: ah - I forgot to push a recent commit

File: setup.py (hunk 5 above)
Comment (LukeWood) [reply setup.py:37]: added

File: setup.py (hunk 4 above)
Comment (LukeWood) [reply setup.py:29]: Done

File: setup.py (hunk 3 above)
Comment (LukeWood) [reply setup.py:22]: done

File: .gitignore (hunk 2 above)
Comment (LukeWood) [reply .gitignore:11]: done

File: .github/workflows/actions.yml [hunk 8]
```diff
@@ -0,0 +1,35 @@
...
+on:
+  push:
+  pull_request:
+  release:
+    types: [created]
+jobs:
+  format:
+    name: Check the code format
+    runs-on: ubuntu-latest
+    steps:
```
Comment (bhack) [inline .github/workflows/actions.yml:12]: Generally, if we don't have a very specific use case, I suggest to use  to use in `steps` the same official dev environment (image) that we distribute to developers/devcontainers as in:
https://github.com/bhack/tensorflow/pull/2

This will really help to minimize the misalignment risk between the prepared environment where people are coding PR on their own hw resourcers/cloud and the CI env.

Comment (bhack) [reply .github/workflows/actions.yml:12]: Ops too late.. you have already merged this.

Comment (bhack) [reply .github/workflows/actions.yml:12]: P.s. I've fixed the link to the PR.
